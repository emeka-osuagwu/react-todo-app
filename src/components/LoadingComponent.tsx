import React from "react";
import * as RB from "react-bootstrap";

function Component() {
    return <RB.Alert variant={"success"}>loading... please wait!</RB.Alert>;
}

export default Component;
