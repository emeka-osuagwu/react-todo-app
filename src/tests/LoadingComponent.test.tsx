import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import LoadingComponent from "../components/LoadingComponent";

test("renders without crashing", () => {
    const { getByText } = render(<LoadingComponent />);
    const title = getByText("loading... please wait!");
    expect(title).toBeInTheDocument();
});

test("renders same mock up", () => {
    const tree = renderer.create(<LoadingComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
