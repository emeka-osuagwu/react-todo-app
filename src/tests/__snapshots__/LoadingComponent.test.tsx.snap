// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders same mock up 1`] = `
<div
  className="fade alert alert-success show"
  role="alert"
>
  loading... please wait!
</div>
`;
